apps/pkcs12.c

```diff
@@ -838,6 +841,51 @@ int pkcs12_main(int argc, char **argv)
     return ret;
 }
 
+static int jdk_trust(PKCS12_SAFEBAG *bag, void *unused)
+{
+    CONF *conf = NULL;
+    const CONF_VALUE *val;
+    STACK_OF(CONF_VALUE) *cb_sk = NULL;
+    const char *cb_attr = NULL;
+    int i;
+    ASN1_OBJECT *obj;
+    X509_ATTRIBUTE *attr;
+    STACK_OF(X509_ATTRIBUTE) *attrs;
+    int ret = 1;
+
+    /* Load the config file */
+    if ((conf = app_load_config(default_config_file)) == NULL)
+        return ret;
+    if (!app_load_modules(conf))
+        goto end;
+
+    attrs = PKCS12_SAFEBAG_get0_attrs(bag);
+
+    /* Let's get the config section we are using */
+    if ((cb_attr = NCONF_get_string(conf, "pkcs12", "certBagAttr")) != NULL) {
+        if ((cb_sk = NCONF_get_section(conf, cb_attr)) != NULL) {
+            for (i = 0; i < sk_CONF_VALUE_num(cb_sk); i++) {
```

Comment:
Yeah I was in two minds about this, you could argue that reading the conf and pulling out the appropriate info is related to the user defined callback. There's no point reading anything in the conf if then you're going to be not setting up the callback so that's why I tried to do it all in the callback but happy to change it if you'd prefer

Reply:
Well - the callback will be called for every cert. So that means you're going to load the config file and parse it for every single cert - which just feels wrong to me.